src/client/account-runner.js

```diff
@@ -294,31 +294,44 @@
         liveDoubleWatch() {
             if (this.usable === false) return null;
 
-            return Bilibili.liveTaskInfo(this.session).then(resp => {
-
-                const doubleStatus = resp['data']['double_watch_info']['status'];
-                const awards = resp['data']['double_watch_info']['awards'];
-                let result = null;
-
-                if (doubleStatus === 1) {
-                    Bilibili.liveDoubleWatch(this.session).then(resp => {
-                        const code = resp['code'];
-                        const msg = resp['msg'] || resp['message'] || '';
-                        if (code !== 0) {
-                            const failedText = `双端奖励领取失败: (${code})${msg}`;
-                            return Promise.reject(failedText);
+            const execute = async () => {
+                let done = false;
+                while (!done) {
+                    Bilibili.liveTaskInfo(this.session).then(resp => {
+
+                        const doubleStatus = resp['data']['double_watch_info']['status'];
+                        const awards = resp['data']['double_watch_info']['awards'];
+
+                        if (doubleStatus === 1) {
+                            done = true;
+                            Bilibili.liveDoubleWatch(this.session).then(resp => {
+                                const code = resp['code'];
+                                const msg = resp['msg'] || resp['message'] || '';
+                                if (code !== 0) {
+                                    const failedText = `双端观看奖励领取失败: (${code})${msg}`;
+                                    cprint(failedText, colors.red);
+                                } else {
+                                    const awardTexts = awards.map(award => `${award.name}(${award.num})`);
+                                    const awardText = '双端观看奖励: ' + awardTexts.join('   ');
+                                    cprint(awardText, colors.green);
+                                }
+                            });
+                        } else if (doubleStatus === 2) {
+                            done = true;
+                            const awardTexts = awards.map(award => `${award.name}(${award.num})`);
+                            const awardText = '双端观看奖励已领取: ' + awardTexts.join('   ');
+                            cprint(awardText, colors.grey);
+                        } else if (doubleStatus === 0) {
+                            cprint('双端观看未完成', colors.green);
                         }
```

Comment:
`Bilibili.liveDoubleWatch` and `Bilibili.liveTaskInfo` can both reject with error. 
In the case of getting a reject, `done` should be set to true to prevent infinite loop. 
Since the call to both methods are inside `async`, a better approach might be:
```javascript
try {
    const taskResp = await Bilibili.liveTaskInfo(...);
    // ...
    const doubleWatchResp = await Bilibili.liveDoubleWatch(...);
    // ...
} catch (error) {
    // ...
    done = true;
}
```

I'll merge the pull and make changes on my end. Thank you for proposing the changes!